Mathlib/RingTheory/FiniteType.lean

```diff
@@ -706,35 +705,102 @@ end MonoidAlgebra
 
 end MonoidAlgebra
 
+section Orzech
+
+open Submodule in
+/-- **Orzech's theorem** for finitely generated module: if `R` is a commutative ring,
+`M` and `N` are `R`-modules, `M` is finitely generated, `i : N →ₗ[R] M` is injective,
+`f : N →ₗ[R] M` is surjective, then `f` is also injective.
+This is a consequence of Noetherian case
+(`IsNoetherian.injective_of_surjective_of_injective`), which requires that `M` is a
+Noetherian module, but allows `R` to be non-commutative. The proof of this result from
+Noetherian case is adapted from <https://math.stackexchange.com/a/1066110>.
+See also Orzech's original paper: *Onto endomorphisms are isomorphisms* [orzech1971]. -/
+theorem Module.Finite.injective_of_surjective_of_injective
+    {R M N : Type*} [CommRing R] [AddCommGroup M] [Module R M] [Module.Finite R M]
+    [AddCommGroup N] [Module R N] (i f : N →ₗ[R] M)
+    (hi : Function.Injective i) (hf : Function.Surjective f) : Function.Injective f := by
+  refine LinearMap.ker_eq_bot.1 <| LinearMap.ker_eq_bot'.2 fun n hn ↦ ?_
```

Comment:
I's a matter of style, but I would have liked to guess what the terse code does.
Why not add a few comments such as "we reduce to the noetherian case…"
In particular, when the induction begins, it's almost impossible to guess what it will be about.

Reply:
> Why not add a few comments such as "we reduce to the noetherian case…"
> In particular, when the induction begins, it's almost impossible to guess what it will be about.

Let me write the sketch in the docstring.

Reply:
Done.